File: material-elements/java/com/zeoflow/material/elements/color/OrientationStrategy.java
package com.zeoflow.material.elements.color;

import android.graphics.PointF;
import android.graphics.Rect;
import android.graphics.drawable.GradientDrawable;
import android.view.MotionEvent;

public interface OrientationStrategy
{

    GradientDrawable.Orientation getGradientOrientation();

    Rect measure(ColorSeekBar view, int widthSpec, int heightSpec);

    Rect calculateGradientBounds(ColorSeekBar view);

    float calculateOffsetOnMotionEvent(ColorSeekBar view, MotionEvent event, Rect barBounds);

    PointF calculateThumbCoordinates(ColorSeekBar view, Rect barBounds);

}

File: material-elements/java/com/zeoflow/material/elements/color/OrientationStrategy.kt
package com.zeoflow.material.elements.color

import android.graphics.PointF
import android.graphics.Rect
import android.graphics.drawable.GradientDrawable
import android.view.MotionEvent

internal interface OrientationStrategy {

  val gradientOrientation: GradientDrawable.Orientation

  fun measure(view: ColorSeekBar, widthSpec: Int, heightSpec: Int): Rect

  fun calculateGradientBounds(view: ColorSeekBar): Rect

  fun calculateOffsetOnMotionEvent(view: ColorSeekBar, event: MotionEvent, barBounds: Rect): Float

  fun calculateThumbCoordinates(view: ColorSeekBar, barBounds: Rect): PointF
}
